frontend/src/api/prometheus/kservePerformanceMetrics.ts

```diff
@@ -178,6 +178,291 @@ export const useFetchKserveMemoryUsageData = (
   });
 };
 
+// Graph #1 - KV Cache usage over time
+type KVCacheUsageData = {
+  data: {
+    kvCacheUsage: PendingContextResourceData<PrometheusQueryRangeResultValue>;
+  };
+  refreshAll: () => void;
+};
+
+export const useFetchKserveKVCacheUsageData = (
+  metricsDef: KserveMetricGraphDefinition,
```

Comment:
```suggestion
  metricsDef: KserveMetricGraphDefinition | NimMetricGraphDefinition,
```

I don't think we need to duplicate files (`frontend/src/api/prometheus/NimPerformanceMetrics.ts` and `frontend/src/api/prometheus/kservePerformanceMetrics.ts`) with just a different `metricsDef` type. Can we remove one of them and union the `metricsDef` type to contain both?